### PR TITLE
CORE-4857 - add description to swagger for file uploads

### PR DESCRIPTION
**Pull request by conal-smith**
EG1:

```
  /fileupload/upload:
    post:
      tags:
        - TestFileUploadAPI
      description: ''
      operationId: post_fileupload_upload
      parameters: []
      requestBody:
        description: requestBody
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                file:
                  type: string
                  description: A file to upload.
                  format: binary
                  nullable: false
                  example: No example available for this type
        required: true
```

EG2:

```
  /fileupload/fileuploadobjectlist:
    post:
      tags:
        - TestFileUploadAPI
      description: ''
      operationId: post_fileupload_fileuploadobjectlist
      parameters: []
      requestBody:
        description: requestBody
        content:
          multipart/form-data:
            schema:
              type: object
              properties:
                files:
                  uniqueItems: false
                  type: array
                  nullable: false
                  items:
                    type: string
                    description: A file to upload.
                    format: binary
                    nullable: false
                    example: No example available for this type
```

### CODE REVIEW

File: libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/SchemaModelProvider.kt [hunk 1]
```diff
@@ -168,7 +168,7 @@ internal class DefaultSchemaModelProvider(private val schemaModelContextHolder:
 
     private fun SchemaModel.applyMetaInfo(param: EndpointParameter) = this.apply {
         this.name = param.name
-        this.description = param.description
+        if(param.description.isNotBlank()) this.description = param.description
```
Comment (sprintengine) [inline libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/SchemaModelProvider.kt:171]: Only overwrite the description on the SchemaModel when there is a description from the annotation

File: libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/builders/SchemaPrimitiveBuilders.kt [hunk 2]
```diff
@@ -108,7 +108,7 @@ internal class SchemaHttpFileUploadBuilder : SchemaBuilder {
         SchemaModel(
             DataType.STRING,
             DataFormat.BINARY
-        )
+        ).apply { description = "A file to upload." }
```
Comment (sprintengine) [inline libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/builders/SchemaPrimitiveBuilders.kt:111]: We add a description to file uploads on the SchemaModel here.

Comment (sprintengine) [reply libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/builders/SchemaPrimitiveBuilders.kt:111]: Could change to just "File upload" WDYT?

Comment (vkolomeyko) [reply libs/http-rpc/http-rpc-server-impl/src/main/kotlin/net/corda/httprpc/server/impl/apigen/processing/openapi/schema/builders/SchemaPrimitiveBuilders.kt:111]: Should be fine.

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Pull request merged**